test/integration/api/FactoryAPI.spec.ts

```diff
@@ -388,4 +391,234 @@ describe('FactoryAPI', () => {
+          new BigNumber(0.2369245),
+          new BigNumber(0.1314204),
+          new BigNumber(0.0415588),
+          new BigNumber(0.1395683),
+          new BigNumber(0.1120511),
+          new BigNumber(0.0939489),
+          new BigNumber(0.0879298),
+          new BigNumber(0.0636428),
+          new BigNumber(0.0558862),
+          new BigNumber(0.0370692),
+        ];
+
+        subjectTargetSetPrice = new BigNumber(100);
+      });
+
+      test('should calculate the correct required component units', async () => {
+        const expectedResult = [
+          new BigNumber('6711742209631728045.33'),
+          new BigNumber('4940616541353383458.65'),
+          new BigNumber('44258572949946751863.68'),
+          new BigNumber('1301943097014925373.13'),
+          new BigNumber('92988464730290.45643154'),
+          new BigNumber('2973066455696202531.65'),
+          new BigNumber('6763830769230769230.77'),
+          new BigNumber('47212759.643916913947'),
+          new BigNumber('140347061778001004.52'),
+          new BigNumber('12760481927710843373.49'),
+        ];
+
+        const result = await subject();
+
+        expect(JSON.stringify(result)).to.equal(JSON.stringify(expectedResult));
+      });
+    });
+  });
+
+  describe('calculateComponentUnits', async () => {
+    let subjectNaturalUnit: BigNumber;
+    let subjectRequiredComponentUnits: BigNumber[];
+
+    function subject(): BigNumber[] {
+      return factoryAPI.calculateComponentUnits(subjectNaturalUnit, subjectRequiredComponentUnits);
+    }
+
+    describe('when the inputs represent a real world integration case', async () => {
+      beforeEach(async () => {
+        subjectRequiredComponentUnits = [
+          new BigNumber('6711742209631728045.33'),
+          new BigNumber('4940616541353383458.65'),
+          new BigNumber('44258572949946751863.68'),
+          new BigNumber('1301943097014925373.13'),
+          new BigNumber('92988464730290.45643154'),
+          new BigNumber('2973066455696202531.65'),
+          new BigNumber('6763830769230769230.77'),
+          new BigNumber('47212759.643916913947'),
+          new BigNumber('140347061778001004.52'),
+          new BigNumber('12760481927710843373.49'),
+        ];
+
+        subjectNaturalUnit = new BigNumber(100000000000);
+      });
+
+
+      test('should calculate the correct component units', async () => {
+        const expectedResult = [
+          new BigNumber('671174220964'),
+          new BigNumber('494061654136'),
+          new BigNumber('4425857294995'),
+          new BigNumber('130194309702'),
+          new BigNumber('9298847'),
+          new BigNumber('297306645570'),
+          new BigNumber('676383076924'),
+          new BigNumber('5'),
+          new BigNumber('14034706178'),
+          new BigNumber('1276048192772'),
+        ];
+
+        const result = await subject();
+
+        expect(JSON.stringify(result)).to.equal(JSON.stringify(expectedResult));
+      });
+    });
+
```

Comment:
You always have this awkward space at the bottom. Is it automatic in your sublime?